### PR TITLE
WiFi Transport: Sharing SSID and password with Mobile Proxy

### PR DESCRIPTION
**Pull request by ashwink11**
This proposal describes a way HMI can share WiFi credentials with mobile proxy when SDL is using WiFi transport with primary transport.

### CODE REVIEW

**Comment by ashwink11** [comment]
@theresalech This PR is ready for review

File: proposals/NNNN-sharing-wifi-ssid-and-password.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    NEHotspotConfiguration * config = [[NEHotspotConfiguration alloc] initWithSSID:ssid passphrase:password isWEP:NO];
+    [config setJoinOnce:NO];
+    
+    NEHotspotConfigurationManager* configManager = [NEHotspotConfigurationManager sharedManager];
+    [configManager applyConfiguration:config completionHandler:^(NSError * _Nullable error) {
+        [self showAlert:[error localizedDescription]];
+        NSLog(@"Error %@",[error debugDescription]);
+    }];
+```
+Capabilities required to use above mentioned API's are as follows:
```
Comment (theresalech) [inline proposals/NNNN-sharing-wifi-ssid-and-password.md]:
```suggestion
Capabilities required to use above mentioned APIs are as follows:
```

File: proposals/NNNN-sharing-wifi-ssid-and-password.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+```
+Hotspot Configuration
+```
+  
+
+## Alternatives considered
+
+None
```
Comment (theresalech) [inline proposals/NNNN-sharing-wifi-ssid-and-password.md]: If possible, can you please complete this section? Per the PR template:
>Describe alternative approaches to addressing the same problem, and why you chose this approach instead.

File: proposals/NNNN-sharing-wifi-ssid-and-password.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
+# WiFi Transport: Sharing SSID and password with Mobile Proxy
+
+* Proposal: [SDL-NNNN](NNNN-sharing-wifi-ssid-and-password)
+* Author: [Ashwin Karemore](https://github.com/ashwink11)
+* Status: **Awaiting review**
+* Impacted Platforms: [Java Suite / iOS / RPC / Core]
```
[the remark this answers is not in the document]
Comment (ashwink11) [reply proposals/NNNN-sharing-wifi-ssid-and-password.md]: Yes, added now.

File: proposals/NNNN-sharing-wifi-ssid-and-password.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+```
+
+On receiving `Transport Event Update` with WiFi credentials, Proxy should check if the mobile device is already connected to WiFi access point mentioned in `Transport Event Update`. If the mobile device is not already connected, it should try connecting to the WiFi access point. 
+
+On successful connection or if the mobile device is already connected to WiFi access point mentioned in `Transport Event Update`, core and proxy should follow use case described in a proposal [Supporting simultaneous multiple transports](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0141-multiple-transports.md "Supporting simultaneous multiple transports").
+
+
+## Potential downsides
+
+None
```
[the remark this answers is not in the document]
Comment (ashwink11) [reply proposals/NNNN-sharing-wifi-ssid-and-password.md]: Added now.

File: proposals/NNNN-sharing-wifi-ssid-and-password.md (hunk 2 above)
Comment (ashwink11) [reply proposals/NNNN-sharing-wifi-ssid-and-password.md]: Added now.

**Comment by ashwink11** [comment]
Thank you for your review @theresalech. I made the required changes.

**Pull request merged**